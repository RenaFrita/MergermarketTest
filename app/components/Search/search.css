.search-form {
  display: inline-block;
  position: relative;
}

.search-label {
  cursor: pointer;
  left: 10px;
  position: absolute;
  top: 10px;
  width: 15px;
}

.search-input {
  border: 2px solid #000;
  height: 20px;
  padding: 6px 25px;
}

.search-clear {
  background: transparent;
  border: 0;
  cursor: pointer;
  padding: 0;
  position: absolute;
  right: 10px;
  top: 10px;
  width: 15px;
}
