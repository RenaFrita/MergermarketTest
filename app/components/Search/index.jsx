import React from 'react';
import PropTypes from 'prop-types';
import search from '../../../assets/search.svg';
import clear from '../../../assets/clear.svg';
import './search.css';

const Search = ({ query, reset, queryString}) => (
  <form className="search-form">
    <label
      htmlFor="search"
      dangerouslySetInnerHTML={{__html: search}}
      className="search-label"
    ></label>
    <input
      className="search-input"
      type="text"
      id="search"
      onChange={query}
      value={queryString}
    />
    <button
      className="search-clear"
      dangerouslySetInnerHTML={{__html: clear}}
      type="button"
      onClick={reset}
    >
    </button>
  </form>
);

Search.propTypes = {
  query: PropTypes.func.isRequired,
  reset: PropTypes.func.isRequired,
  queryString: PropTypes.string.isRequired
};

export default Search;
