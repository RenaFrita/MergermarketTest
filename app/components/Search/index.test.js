import React from 'react';
import Search from './index.jsx';
import renderer from 'react-test-renderer';

test('Search component markup', () => {
  const mockFn = jest.fn();
  const component = renderer.create(
    <Search
      query={mockFn}
      reset={mockFn}
      queryString={''}
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
