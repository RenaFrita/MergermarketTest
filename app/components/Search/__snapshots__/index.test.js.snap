// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component markup 1`] = `
<form
  className="search-form"
>
  <label
    className="search-label"
    dangerouslySetInnerHTML={
      Object {
        "__html": "test-file-stub",
      }
    }
    htmlFor="search"
  />
  <input
    className="search-input"
    id="search"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="search-clear"
    dangerouslySetInnerHTML={
      Object {
        "__html": "test-file-stub",
      }
    }
    onClick={[Function]}
    type="button"
  />
</form>
`;
